Automattic/wp-calypso
# Subscription management: add data fetch queries and conditional logic

Closes https://github.com/Automattic/wp-calypso/issues/74508
Closes https://github.com/Automattic/wp-calypso/issues/74422

## Proposed Changes

This PR adds:
- A `fetchFromApi` helper function, that fetches using the wpcom-request-proxy for logged in users and apiFetch for external users. In the case of external users it adds the subkey cookie to the Authorization header, as required by the API
- It adds a `useSubscriptionManagerUserSettingsQuery()`, utilizing `react-query`
- It moves API types to `@automattic/data-stores`

## Testing Instructions

- Checkout this branch
- For testing, open up a component inside `packages/subscription-manager` and add the hook. This is for example the hook applied in `UserSettings.tsx`:

```js
import type { SubscriptionManagerUserSettings } from '@automattic/data-stores';
import { useSubscriptionManagerUserSettingsQuery } from "@automattic/data-stores";

type UserSettingsProps = {
	value?: SubscriptionManagerUserSettings;
	onChange?: ( value: SubscriptionManagerUserSettings ) => void;
	loading?: boolean;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
const UserSettings = ( { value = {}, loading = false }: UserSettingsProps ) => {
	const { data } = useSubscriptionManagerUserSettingsQuery();
	console.log(data);
	return <div className="user-settings">User settings will go here</div>
}

export default UserSettings;
```
- Visit this page logged in
- In the console you should see the API response (the initialData should be `{}`)

Now switch to a incognito window & paste this in the console:

- `document.cookie = 'subkey=b8bb71801b61d6a7c7ec682aaa8af4e1.jefpober+testetst@gmail.com';`
- Visit the same page
- In the console you should see the API response (the initialData should be `{}`)

## Todo
There are many optimizations that can still be done, but this is a first working copy that should help unblock Calypso development on this feature.

Todo:
- [ ] Utilize `QueryClientProvider` and `QueryClient` for better cache control
- [ ] Better cookie parsing 

## Pre-merge Checklist
